library/src/components/Schema.tsx

```diff
@@ -38,17 +38,31 @@ export const Schema: React.FunctionComponent<Props> = ({
     return null;
   }
 
-  const uid = schema.uid();
   const dependentSchemas = SchemaHelpers.getDependentSchemas(schema);
 
   const constraints = SchemaHelpers.humanizeConstraints(schema);
-  const isExpandable = SchemaHelpers.isExpandable(schema) || dependentSchemas;
   const externalDocs = schema.externalDocs();
 
   const renderType = schema.ext(SchemaHelpers.extRenderType) !== false;
   const rawValue = schema.ext(SchemaHelpers.extRawValue) === true;
   const parameterLocation = schema.ext(SchemaHelpers.extParameterLocation);
 
+  const isExpandable = SchemaHelpers.isExpandable(schema) || dependentSchemas;
+  isCircular = isCircular || schema.ext('x-parser-circular') || false;
+
+  let uid = schema.uid();
+
+  // checking uid for circular items
+  // after fixing https://github.com/asyncapi/parser-js/issues/293 statement should be removed
+  if (
+    isCircular &&
+    !uid &&
+    schema.items() &&
+    typeof (schema.items() as SchemaType).uid === 'function'
```

Comment:
how come uid could be a function?

Reply:
Items can be also array of schemas, like tuple type, so for that I used `typeof (schema.items() as SchemaType).uid === 'function'` to check that I operate on schema, not on array of schemas.

Reply:
Also `uid` is used here to check what exactly schema is circular

Reply:
> Items can be also array of schemas, like tuple type, so for that I used typeof (schema.items() as SchemaType).uid === 'function' to check that I operate on schema, not on array of schemas.

this is not that straight forward, can you add a comment to code please 🙏🏼 

> Also uid is used here to check what exactly schema is circular

I don't know what you mean. Tbh I don't remember how it is handled on the parser side. You mean that circular schema has no `uid`?

Reply:
> You mean that circular schema has no uid?

Circular schemas have uid, but problem with `items` is that `uid` is added to schema in `items` field but `x-parser-circular` info is added to parent schema, which has `items` field 😅 So I must do this "tricky" think here - I have info that schema has circular ref in items, but I must read uid from `items()` function

Reply:
🤯 
this one definitely deserves a code comment, way to deep knowledge about the parser needed to get the code 😄

Reply:
Added :)